### ISSUE TITLE
Possible incorrect electron temperature values in Langmuir diagnostics

### ISSUE THREAD

**Issue by LeoMurphyWM24** [PlasmaPy/PlasmaPy#1483]
The `swept_probe_analysis` function in the Langmuir diagnostics module returns non-bimaxwellian temperatures that are the reciprocal of what I expect.
I apologize if I make mistakes in details or formatting. I have three main reasons for my suspicion:
1. The example on the [Langmuir probe data analysis page](https://docs.plasmapy.org/en/stable/notebooks/diagnostics/langmuir_analysis.html) using data from David Pace analyzes a characteristic using both maxwellian (`bimaxwellian=False`) and bimaxwellian (`bimaxwellian=True`) methods. 
   - In the first case, the calculated electron temperature is 0.32266985 eV.
   - In the second case, the calculated cold and hot electron temperatures are 3.146203 eV and 3.72242199 eV, respectively. 
   The reciprocal of 0.32267 is 3.099, which is much closer to the two bimaxwellian temperatures than 0.32267 is. [David Pace’s data](https://davidpace.com/wp-content/uploads/2015/11/sampleIV.csv) says that his calculated electron temperature is 3.57 eV.
2. I ran `swept_probe_analysis` on a 2D array of characteristics and made a contour plot of the electron temperature. We expected electron temperatures in the range of 2-5 eV, but got non-bimaxwellian electron temperatures in the range of 0.2-0.5 eV. The bimaxwellian electron temperatures were in the range we expected.
   - ![Average bimaxwellian temperature](https://user-images.githubusercontent.com/85201894/158479532-b3490661-e126-454b-9f84-a68d60b15bd9.png)
   A contour plot of bimaxwellian electron temperatures. This looks similar to the expected temperatures in the last image.
   - ![Maxwellian electron temperature](https://user-images.githubusercontent.com/85201894/158479595-8dba5f67-f456-4616-a5fe-70039cdcd93f.png) 
   A contour plot of maxwellian electron temperatures. This looks very roughly like an inverted version of the first and third plots.
   - <img width="504" alt="T_e_MATLAB" src="https://user-images.githubusercontent.com/85201894/158479754-ba23c2a4-3042-402d-b823-d3607b9f6c04.png">
    A contour plot of maxwellian electron temperature calculated using outside code. It supports the bimaxwellian temperatures in the first image.
I unfortunately don’t have a minimum reproducible code snippet to recreate these graphs.
3. Inside [Langmuir.py](https://github.com/PlasmaPy/PlasmaPy/blob/main/plasmapy/diagnostics/langmuir.py), the `get_electron_temperature` function calls `_fit_func_lin_inverse` in the maxwellian case and `_fit_func_double_lin_inverse` in the bimaxwellian case. Both fit functions take “inverse slope parameters”, according to the docstring. The `curve_fit` function returns these parameters, which are inversely proportional to the sweep characteristic slope.  
In the bimaxwellian case, the `get_electron_temperature` function returns `T0` and `T0 + Delta_T` in units of eV. But in the maxwellian case (`bimaxwellian=False`), `get_electron_temperature` assigns the value of `T0` to a variable named `slope`, even though `T0` is the inverse of the slope, and then returns the reciprocal of this:
```
if not bimaxwellian:
        slope = fit[2]

        T_e = slope ** -1 * u.eV
```
If I understand the electron temperature calculation, this should be one more reciprocal than necessary - the returned temperature value is now inversely proportional to `T0` and directly proportional to the slope, while the `get_electron_temperature` docstring says that “the slope is inversely proportional to the temperature”. 

I’m still fairly new to PlasmaPy and this is my first issue submission, so please let me know if I have anything wrong. I can provide more details wherever helpful.

**Comment by rocco8773** [comment]
@LeoMurphyWM24 You are correct about these lines being incorrect...

https://github.com/PlasmaPy/PlasmaPy/blob/f39db36b1daeb146d79a148c89ecd974e79eeddd/plasmapy/diagnostics/langmuir.py#L941-L944

Because of how `_fit_func_lin_inverse` is defined and how `scipy.optimize.curve_fit` work,  the returned value `fit[2]` is the electron temperature.  So, the current code is improperly inverting it.

Do you mind adding this fix the PR (#1487) you just opened about the other bug in this function?

**Comment by LeoMurphyWM24** [comment]
Hi @rocco8773, thanks for your response. I can add this fix to PR #1487. 
To respond to your comments:

1. My characteristics go from a negative bias and negative ion saturation current to a positive bias and positive electron saturation current, so matching the orientation of the [example characteristics](https://docs.plasmapy.org/en/stable/notebooks/diagnostics/langmuir_analysis.html) and the negative of the actual current.

2. Thanks for the clarification. I used `swept_probe_analysis` to get my electron temperature statistics this time, but I'll keep this in mind in working with `get_electron_temperature`.

**Comment by rocco8773** [comment]
Closed by #1487
